components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/test/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorServiceTestCase.java

```diff
@@ -136,5 +138,61 @@ public void testRevokeAccessToken() throws Exception {
+        accessTokenInfo.setScopes("apim:workflow_approve apim:subscribe openid");
+
+        //// Expected AuthResponseBean object
+        AuthResponseBean expectedAuthResponseBean = new AuthResponseBean();
+        expectedAuthResponseBean.setTokenValid(true);
+        expectedAuthResponseBean.setAuthUser("admin");
+        expectedAuthResponseBean.setScopes(accessTokenInfo.getScopes());
+        expectedAuthResponseBean.setType(AuthenticatorConstants.BEARER_PREFIX);
+        expectedAuthResponseBean.setValidityPeriod(accessTokenInfo.getValidityPeriod());
+        expectedAuthResponseBean.setIdToken(accessTokenInfo.getIdToken());
+
+        KeyManager keyManager = Mockito.mock(KeyManager.class);
+        AuthenticatorService authenticatorService = new AuthenticatorService(keyManager);
+
+        //// Actual response
+        AuthResponseBean authResponseBean = new AuthResponseBean();
+        authResponseBean = authenticatorService.setAccessTokenData(authResponseBean, accessTokenInfo);
+        Assert.assertTrue(EqualsBuilder.reflectionEquals(expectedAuthResponseBean,authResponseBean));
+
+        // Happy Path - When id token is null
+        //// AccessTokenInfo object with null id token
+        AccessTokenInfo invalidTokenInfo = new AccessTokenInfo();
+        invalidTokenInfo.setValidityPeriod(-2L);
+        invalidTokenInfo.setScopes("apim:workflow_approve apim:subscribe openid");
+
+        //// Expected AuthResponseBean object when id token is null
+        AuthResponseBean expectedResponseBean = new AuthResponseBean();
+        expectedResponseBean.setTokenValid(true);
+        expectedResponseBean.setScopes(invalidTokenInfo.getScopes());
+        expectedResponseBean.setType(AuthenticatorConstants.BEARER_PREFIX);
+        expectedResponseBean.setValidityPeriod(invalidTokenInfo.getValidityPeriod());
+        expectedResponseBean.setIdToken(invalidTokenInfo.getIdToken());
+
+        //// Actual response when id token is null
+        AuthResponseBean responseBean = new AuthResponseBean();
+        authenticatorService.setAccessTokenData(responseBean, invalidTokenInfo);
+        Assert.assertTrue(EqualsBuilder.reflectionEquals(expectedResponseBean, responseBean));
+
+        // Error Path - When parsing JWT fails and throws KeyManagementException
+        //// AccessTokenInfo object with invalid ID token format
+        AccessTokenInfo invalidAccessTokenInfo = new AccessTokenInfo();
+        invalidAccessTokenInfo.setIdToken("xxx-invalid-id-token-xxx");
+        invalidAccessTokenInfo.setValidityPeriod(-2L);
+        invalidAccessTokenInfo.setScopes("apim:workflow_approve apim:subscribe openid");
+
+        try {
+            AuthResponseBean errorResponseBean = new AuthResponseBean();
+            authenticatorService.setAccessTokenData(errorResponseBean, invalidAccessTokenInfo);
+        } catch (KeyManagementException e) {
+            Assert.assertEquals(e.getMessage(), "JWT Parsing failed. Invalid JWT.");
```

Comment:
We should find a better way of asserting this. If someone changes the message in the exception, this test case will fail. Which is not the ideal behavior.

Reply:
This has been fixed in https://github.com/wso2/carbon-apimgt/pull/5106